episodes/excthreesteps.Rmd

```diff
@@ -0,0 +1,517 @@
+            dispDefField(def_field, spacing=disp_spacing, plot_type=disp_method_df)
+            axs_combined[1].set_title('Deformation Field')
+
+        else:
+            [jacobian, _] = calcJacobian(def_field)
+            dispImage(jacobian, title='Jacobian')
+            plt.set_cmap('jet')
+            #plt.colorbar()
+
+        axs_combined[2].clear()
+        plt.sca(axs_combined[2])
+        diff_image = images[current_image_index[0]] - target
+        dispImage(diff_image, title='Difference Image')
+
+        fig_combined.canvas.draw()
+
+
+    # Create a single figure with 3 subplots
+    fig_combined, axs_combined = plt.subplots(1, 3, figsize=(12, 6))
+
+    # Display initial images
+    plt.sca(axs_combined[0])
+    dispImage(images[current_image_index[0]], title=image_titles[current_image_index[0]])
+
+    plt.sca(axs_combined[1])
+    dispDefField(def_field, spacing=disp_spacing, plot_type=disp_method_df)
+    axs_combined[1].set_title('Deformation Field')
+
+    plt.sca(axs_combined[2])
+    diff_image = images[current_image_index[0]] - target
+    dispImage(diff_image, title='Difference Image')
+
+    # Add instructions for navigating images
+    fig_combined.text(0.5, 0.02, 'Press <- or -> to navigate between source, target and warped images, Press Up or Down to switch between deformation field and Jacobian', ha='center', va='top', fontsize=12, color='black')
+
+    # Connect the key event handler to the figure
+    fig_combined.canvas.mpl_connect('key_press_event', on_key)
+
+    plt.tight_layout()
+    plt.show()
+```
+
+
+## 12. Preparing Images for calling the demonsReg function
+
+Follow `exampleSolution3.py` available in `src/mirc/Example` for running the function. Below, we describe how we prepare images: `cine_MR_img_1.png`, `cine_MR_img_2.png`, and `cine_MR_img_3.png`. These images are available in `episodes/data`. 
+
+
+```python
+import skimage.io
+cine_MR_img_1 = skimage.io.imread('path/to/your/image/../cine_MR_1.png') 
+cine_MR_img_2 = skimage.io.imread('path/to/your/image/../cine_MR_2.png')
+cine_MR_img_3 = skimage.io.imread('path/to/your/image/../cine_MR_3.png')
+```
+
+Converting all images into double data for standard orientation.
+
+```python
+import numpy as np
+# ***************
+# ADD CODE HERE TO CONVERT ALL THE IMAGES TO DOUBLE DATA TYPE AND TO REORIENTATE THEM
+# INTO 'STANDARD ORIENTATION'
+cine_MR_img_1 = np.double(cine_MR_img_1)
+cine_MR_img_2 = np.double(cine_MR_img_2)
+cine_MR_img_3 = np.double(cine_MR_img_3)
+
+cine_MR_img_1 = np.flip(cine_MR_img_1.T, 1)
+cine_MR_img_2 = np.flip(cine_MR_img_2.T, 1)
+cine_MR_img_3 = np.flip(cine_MR_img_3.T, 1)
+```
+
+
+### Calling the function:
+
+```python
+demonsReg(cine_MR_img_1, cine_MR_img_2)
+```
+
+### Final Results Video (Navigation between images and modes)
+
+<video width="800" height="450" controls>
+    <source src="fig/final_results_vid.mp4" type="video/mp4">
+    Your browser does not support the video tag.
+</video>
```

Comment:
Would you like to add a caption to the video and description of what the user is looking in the video?

Reply:
a title is added for that